pwiz_tools/Skyline/Model/Proteome/ProteinGroupMetadata.cs

```diff
@@ -0,0 +1,216 @@
+
+        public override ProteinMetadata ChangeName(string name)
+        {
+            return new ProteinGroupMetadata(this);
+        }
+
+        public override ProteinMetadata ChangeDescription(string descr)
+        {
+            return new ProteinGroupMetadata(this);
+        }
+
+        public override ProteinMetadata ChangePreferredName(string preferredname)
+        {
+            return new ProteinGroupMetadata(this);
+        }
+
+        public override ProteinMetadata ChangeAccession(string accession)
+        {
+            return new ProteinGroupMetadata(this);
+        }
+
+        public override ProteinMetadata ChangeGene(string gene)
+        {
+            return new ProteinGroupMetadata(this);
+        }
+
+        public override ProteinMetadata ChangeSpecies(string species)
+        {
+            return new ProteinGroupMetadata(this);
+        }
+
+        public ProteinMetadata ChangeSingleProteinMetadata(ProteinMetadata singleProteinMetadata)
+        {
+            Assume.IsTrue(singleProteinMetadata != null && !(singleProteinMetadata is ProteinGroupMetadata));
+            var proteinMetadataList = new List<ProteinMetadata>(ProteinMetadata);
+            var matchingProteinIndex = proteinMetadataList.IndexOf(m => m.Name == singleProteinMetadata?.Name);
+            Assume.IsTrue(matchingProteinIndex >= 0, $@"no matching protein in group {this} for protein {singleProteinMetadata}");
+            proteinMetadataList[matchingProteinIndex] = singleProteinMetadata;
+            return new ProteinGroupMetadata(Name, Description, proteinMetadataList);
+        }
+
+        public override ProteinMetadata ChangeWebSearchInfo(WebSearchInfo webSearchInfo)
+        {
+            return new ProteinGroupMetadata(this, webSearchInfo);
+        }
+
+        public override ProteinMetadata ClearWebSearchInfo()
+        {
+            // sometimes all you really want is to initialize
+            return new ProteinGroupMetadata(this, WebSearchInfo.EMPTY);
+        }
+
+        public override ProteinMetadata SetWebSearchCompleted()
+        {
+            return new ProteinGroupMetadata(this, WebSearchInfo.SetSearchCompleted());
+        }
+
+        public override ProteinMetadata SetWebSearchTerm(WebSearchTerm search)
+        {
+            return new ProteinGroupMetadata(this, WebSearchInfo.SetSearchTerm(search));
+        }
+
+        /// <summary>
+        /// returns a copy of this, with any blanks filled in
+        /// using the members of source
+        /// </summary>
+        /// <param name="source">the source to merge from</param>
+        /// <returns>a copy of this merged with source, with this winning when fields conflict</returns>
+        public override ProteinMetadata Merge(ProteinMetadata source)
+        {
+            if (source == null)
+                return this;
+            if (source is ProteinGroupMetadata sourceGroup)
```

Comment:
A comment explaining would be good. Because the functions Name and Description are virtual, these really appear the same except for how ProteinMetadata gets handled. So you might make that clearer by having just one "return new ProteinGroupMetadata()" call and the if determining only what ProteinMetadata gets used with an explanation why the sourceGroup overrides this when it has ProteinMetadata. Seems obvious whey this.ProteinMetadata gets used when source has none.